src/event.rs

```diff
@@ -99,13 +99,18 @@ impl Event {
                 } else {
                     ReadWrite::Write
                 };
-                #[cfg(linux4_14)]
-                let thread_id = Pid::from_raw(unsafe { pagefault.feat.ptid });
+                // Converting the ptid to i32 is safe because the maximum pid in
+                // Linux is 2^22, which is about 4 million.
+                //
+                // Reference:
+                //   https://github.com/torvalds/linux/blob/2d338201d5311bcd79d42f66df4cecbcbc5f4f2c/include/linux/threads.h
+                #[cfg(feature = "linux4_14")]
+                let thread_id = Pid::from_raw(unsafe { pagefault.feat.ptid } as i32);
```

Comment:
ah i see, we hadn't gotten the type error for passing a u32 into `from_raw(i32) -> Pid` because we hadn't actually compiled in this arm of the code?

i actually think the correctness here is more importantly that `ptid` (`u32`) and `pid_t` (`i32`) are the same number of bits, so casting from one to the other won't be a truncating conversion. even if pids could stride to `2^32-1`, `Pid` should preserve the provided bit pattern through to users' syscalls and things _ought_ to work - so long as nothing truncates the pids themselves.

if that makes sense to you too, it might be worth reframing the comment in terms of [`ptid`'s definition](https://github.com/torvalds/linux/blob/cdeffe87f790dfd1baa193020411ce9a538446d7/include/uapi/linux/userfaultfd.h#L103) and [`libc`'s `pid_t` definition](https://docs.rs/libc/latest/libc/type.pid_t.html). and then hoping that no one changes the signedness of these types between architectures, like i've seen libc do for a few things :)

Reply:
@iximeow That's interesting, I like your argument about bitwise exactness. It makes sense to me, but I'm not 100% sure about the reasoning. Maybe @pchickey has an opinion on this?

Reply:
I don't particularly have an opinion on this. I think @iximeow's additions make sense.

Stepping back though, is there a reason that we're storing `Pid` here in the `Event::Pagefault` variant rather than simply `i32` coming in from the raw bindings?

Reply:
i'm pretty sure the reason we wrap it in a `Pid` is just data hygiene: conceptually the number we got back is a `pid_t`\* and we didn't want to forget about it or easily misuse the value later on in client code, so make it the "most right" type as soon as we could to minimize that risk.

then, `nix::unistd::Pid` is a safe bet for giving this a type, because we'd _probably_ want to use it with something else `nix` wraps, rather than going to the syscalls or libc directly. that said, there's [not much](https://docs.rs/nix/latest/nix/unistd/struct.Pid.html?search=pid) you can do with a `nix::unistd::Pid` directly that isn't some kind of debugging operation, and i think we never did use it for any of this stuff in Lucet.

i think the idea that this is a pid\* and we should type it as such still holds at least - if `nix::unistd::Pid` ever truncates its argument it'd be a very serious bug that must get fixed upstream asap anyway.

\* it's not a pid, but things at like it's a pid, and types-wise, it kiiiinda is... `gettid(2)` for example is `pid_t gettid(void)`. for most things you'd use a pid for (my favorite example is "checking `/proc/<pid>/maps`"), the thread ID returned here "is" a pid. i believe PIDs and TIDs share the same namespace, so it's _pretty_ pid-y... but not a pid. :( this is, however, how it ended up being a `Pid` in `nix`, and eventually a `Pid` here.`